flocker/node/agents/blockdevice.py

```diff
@@ -883,6 +883,8 @@ def destroy_volume(blockdevice_id):
 
         :raises UnknownVolume: If the supplied ``blockdevice_id`` does not
             exist.
+            Note that the ebs and cinder implementation can raise a
```

Comment:
Reinforcing this.  I'd say instead that details about particular implementations of the interface belong on the implementations.  If a consumer of the interface *has* to know about those implementation details in order to use the interface correctly then the implementations don't actually conform to the interface (ie, they're broken).

Maybe the thing to say here instead is that `TimeoutException` is just another exception this method should be expected to raise.  Ideally that would come with tests and some changes to the existing consumers of the interface (so perhaps a follow-up issue).